component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/twitter/TwitterAuthenticator.java

```diff
@@ -71,12 +72,48 @@ public String getContextIdentifier(HttpServletRequest request) {
         return request.getParameter(TwitterAuthenticatorConstants.STATE_PARAM);
     }
 
+    @Override
     public boolean canHandle(HttpServletRequest request) {
 
+        return ((isOauthParamExists(request) && TwitterAuthenticatorConstants.TWITTER_LOGIN_TYPE.equals
+                (request.getParameter(TwitterAuthenticatorConstants.LOGIN_TYPE_PARAM))) || isErrorParamExists(request));
+    }
+
+    /**
+     * Check access denied error param exist in request.
+     *
+     * @param request httpServletRequest
+     * @return true or false
+     */
+    private boolean isErrorParamExists(HttpServletRequest request) {
+        return request.getParameter(TwitterAuthenticatorConstants.OAUTH2_PARAM_ERROR) != null;
+    }
+
+    /**
+     * Check whether oauth_token and oauth_verifier param exist in request.
+     *
+     * @param request httpServletRequest
+     * @return true or false
+     */
+    private boolean isOauthParamExists(HttpServletRequest request) {
         return (request.getParameter(TwitterAuthenticatorConstants.TWITTER_OAUTH_TOKEN) != null
-                && request.getParameter(TwitterAuthenticatorConstants.TWITTER_OAUTH_VERIFIER) != null)
-                && TwitterAuthenticatorConstants.TWITTER_LOGIN_TYPE.equals(request.getParameter
-                (TwitterAuthenticatorConstants.LOGIN_TYPE_PARAM));
+                && request.getParameter(TwitterAuthenticatorConstants.TWITTER_OAUTH_VERIFIER) != null);
+    }
+
+    /**
+     * Handle error response when click on cancel without providing credentials.
+     *
+     * @param request httpServletRequest
+     * @throws InvalidCredentialsException
+     */
+    private void handleErrorResponse(HttpServletRequest request) throws InvalidCredentialsException {
+        if (isErrorParamExists(request)) {
+            String error = request.getParameter(TwitterAuthenticatorConstants.OAUTH2_PARAM_ERROR);
+            if (log.isDebugEnabled()) {
+                log.debug("Failed to authenticate via Twitter. " + error);
```

Comment:
create a meaning full debug log.
log.debug("Failed to authenticate via Twitter: " + error);